cl_manycore/regression/cuda/test_dram.c

```diff
@@ -0,0 +1,157 @@
+
+
+	/*****************************************************************************************************************
+	* Prepare list of input arguments for kernel.
+	******************************************************************************************************************/
+	int argv[2] = {A_ptr_device, block_size_x};
+
+	/*****************************************************************************************************************
+	* Enquque grid of tile groups, pass in grid and tile group dimensions, kernel name, number and list of input arguments
+	******************************************************************************************************************/
+	rc = hb_mc_grid_init (&device, grid_dim, tg_dim, "kernel_dram", 2, argv);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to initialize grid.\n");
+		return HB_MC_FAIL;
+	}
+
+
+	/*****************************************************************************************************************
+	* Launch and execute all tile groups on device and wait for all to finish. 
+	******************************************************************************************************************/
+	rc = hb_mc_device_tile_groups_execute(&device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to execute tile groups.\n");
+		return HB_MC_FAIL;
+	}	
+
+
+
+	/*****************************************************************************************************************
+	* Copy result matrix back from device DRAM into host memory. 
+	******************************************************************************************************************/
+	uint32_t A_ptr_host;
+	void *src = (void *) ((intptr_t) A_ptr_device);
+	void *dst = (void *) &A_ptr_host;
+	rc = hb_mc_device_memcpy (&device, (void *) dst, src, 1 * sizeof(uint32_t), hb_mc_memcpy_to_host); /* copy A_ptr to the host */
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to copy memory from device.\n");
+		return HB_MC_FAIL;
+	}
+
+
+	uint32_t A_host[N];
+	src = (void *) ((intptr_t) A_ptr_host);
+	dst = (void *) &A_host[0];
+	rc = hb_mc_device_memcpy (&device, (void *) dst, src, N * sizeof(uint32_t), hb_mc_memcpy_to_host); /* copy A to the host using A_ptr*/
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to copy memory from device.\n");
+		return HB_MC_FAIL;
+	}
+
+
+
+	/*****************************************************************************************************************
+	* Freeze the tiles and memory manager cleanup. 
+	******************************************************************************************************************/
+	rc = hb_mc_device_finish(&device); 
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to de-initialize device.\n");
+		return HB_MC_FAIL;
+	}
+
+
+	int mismatch = 0; 
+	for (int i = 0; i < N; i++) {
+		if (A_host[i] != i) { 
+			bsg_pr_err(BSG_RED("Mismatch") ": -- A[%d] = 0x%x\t Expected: 0x%x\n", i , A_host[i], i);
```

Comment:
I recommend formatting 32-bit hex numbers like this: `"%08" PRIx32`
1. More important: The `"08"` pads with 0s in the upper digits and the resulting print statements, I find, are much easier to read.
2. Less important, but good to know: The `PRIx32` is defined in `<inttypes.h>` and it addresses the fact that `%x` is for the type of size `unsigned int` or `int` but that does not mean a 32-bit integer on every system.

Reply:
interesting, very nice pointer! I replaced all 0x%x notations with this one, thanks!